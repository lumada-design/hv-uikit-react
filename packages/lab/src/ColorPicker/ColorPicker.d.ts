import { StandardProps } from "@material-ui/core";
import { HvFormElementProps } from "@hitachivantara/uikit-react-core";

export type HvColorPickerClassKey = "root";

export interface HvColorPickerProps
  extends StandardProps<HvFormElementProps, HvColorPickerClassKey, "onChange"> {
  /**
   * The value color, in HEX format.
   */
  value?: string;
  /**
   * The default value color, in HEX format.
   */
  defaultValue?: string;
  /**
   * If `true` the dropdown is disabled unable to be interacted, if `false` it is enabled.
   */
  disabled?: boolean;
  /**
   * If `true` the dropdown starts opened if `false` it starts closed.
   */
  expanded?: boolean;
  /**
   * When uncontrolled, defines the initial expanded state.
   */
  defaultExpanded?: boolean;
  /**
   * A function to be executed whenever the color changes.
   */
  onChange?: (color: string) => void;
}

export default function HvColorPicker(props: HvColorPickerProps): JSX.Element | null;
