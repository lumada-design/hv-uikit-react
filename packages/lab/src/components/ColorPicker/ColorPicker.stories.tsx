import { Meta, StoryObj } from "@storybook/react";
import { HvColorPicker, HvColorPickerProps } from "./ColorPicker";

const meta: Meta<typeof HvColorPicker> = {
  title: "Lab/Color Picker",
  component: HvColorPicker,
};
export default meta;

export const Main: StoryObj<HvColorPickerProps> = {
  args: {
    label: "Color",
  },
  argTypes: {
    classes: { control: { disable: true } },
    description: { control: { disable: true } },
    "aria-label": { table: { disable: true } },
    "aria-describedby": { table: { disable: true } },
    "aria-labelledby": { table: { disable: true } },
  },
  render: (args) => {
    return (
      <div style={{ width: "134px" }}>
        <HvColorPicker {...args} onChange={(color) => console.log(color)} />
      </div>
    );
  },
};

export const WithoutSavedColors: StoryObj<HvColorPickerProps> = {
  parameters: {
    docs: {
      description: {
        story: "ColorPicker component without the saved colors area.",
      },
    },
  },
  render: () => {
    return (
      <div style={{ width: "134px" }}>
        <HvColorPicker
          showSavedColors={false}
          onChange={(color) => console.log(color)}
          defaultValue="#C62828"
        />
      </div>
    );
  },
};

export const OnlyRecommendedColors: StoryObj<HvColorPickerProps> = {
  parameters: {
    docs: {
      description: {
        story: "ColorPicker component with only the recommend colors area.",
      },
    },
  },
  render: () => {
    return (
      <div style={{ width: "134px" }}>
        <HvColorPicker
          showSavedColors={false}
          showCustomColors={false}
          onChange={(color) => console.log(color)}
          defaultValue="#F6941E"
        />
      </div>
    );
  },
};

export const IconOnly: StoryObj<HvColorPickerProps> = {
  parameters: {
    docs: {
      description: {
        story: "ColorPicker component that only shows an icon.",
      },
    },
  },
  render: () => {
    return (
      <div style={{ width: "134px" }}>
        <HvColorPicker
          iconOnly={true}
          onChange={(color) => console.log(color)}
        />
      </div>
    );
  },
};

export const IconOnlyWithoutSavedColors: StoryObj<HvColorPickerProps> = {
  parameters: {
    docs: {
      description: {
        story:
          "ColorPicker component that only shows an icon and without the saved colors area.",
      },
    },
  },
  render: () => {
    return (
      <div style={{ width: "134px" }}>
        <HvColorPicker
          iconOnly={true}
          showSavedColors={false}
          defaultValue="#477DBD"
          onChange={(color) => console.log(color)}
        />
      </div>
    );
  },
};

export const IconOnlyRecommendedColors: StoryObj<HvColorPickerProps> = {
  parameters: {
    docs: {
      description: {
        story:
          "ColorPicker component that only shows an icon and the recommended colors.",
      },
    },
  },
  render: () => {
    return (
      <div style={{ width: "134px" }}>
        <HvColorPicker
          iconOnly={true}
          showSavedColors={false}
          showCustomColors={false}
          defaultValue="#59941B"
          onChange={(color) => console.log(color)}
        />
      </div>
    );
  },
};

export const CustomizedColorPicker: StoryObj<HvColorPickerProps> = {
  parameters: {
    docs: {
      description: {
        story:
          "ColorPicker component with several customizations. With these customizations you can recreate the DS3 component.",
      },
    },
  },
  render: () => {
    return (
      <div style={{ width: "240px" }}>
        <HvColorPicker
          showLabels={false}
          showSavedColors={false}
          dropdownIcon="arrow"
          defaultValue="#EC018B"
          onChange={(color) => console.log(color)}
          recommendedColorsPosition="bottom"
          recommendedColors={[
            "#C62828",
            "#DB6B22",
            "#F6941E",
            "#8DC63D",
            "#59941B",
            "#00838F",
            "#05A99C",
            "#01ADEF",
            "#477DBD",
            "#0155A5",
            "#2E3192",
            "#652C90",
            "#AA00FF",
            "#EC018B",
            "#999999",
            "#414141",
          ]}
        />
      </div>
    );
  },
};
