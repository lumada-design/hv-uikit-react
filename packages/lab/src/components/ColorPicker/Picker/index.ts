export { default as Picker } from "./Picker";
