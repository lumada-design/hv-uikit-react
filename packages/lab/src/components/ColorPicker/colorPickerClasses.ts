import { getClasses } from "@hitachivantara/uikit-react-core";

export interface HvColorPickerClasses {
  root?: string;
  labelContainer?: string;
  label?: string;
  description?: string;
  headerColorIcon?: string;
  headerColorValue?: string;
  panel?: string;
  colorPicker?: string;
  colorPickerIcon?: string;
  dropdownRootIconOnly?: string;
  headerColorIconOnly?: string;
  recommendedColorsRoot?: string;
  pickerFields?: string;
}

const classKeys: (keyof HvColorPickerClasses)[] = [
  "root",
  "labelContainer",
  "label",
  "description",
  "headerColorIcon",
  "headerColorValue",
  "panel",
  "colorPicker",
  "colorPickerIcon",
  "dropdownRootIconOnly",
  "headerColorIconOnly",
  "recommendedColorsRoot",
  "pickerFields",
];

const colorPickerClasses = getClasses(classKeys, "HvColorPicker");

export default colorPickerClasses;
