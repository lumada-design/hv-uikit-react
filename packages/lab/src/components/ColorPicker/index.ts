export { default as colorPickerClasses } from "./colorPickerClasses";
export * from "./colorPickerClasses";
export * from "./ColorPicker";
