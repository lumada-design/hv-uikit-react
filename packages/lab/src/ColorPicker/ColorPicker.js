import React, { useMemo } from "react";
import PropTypes from "prop-types";
import clsx from "clsx";

import { withStyles, useTheme } from "@material-ui/core";
import {
  HvPanel,
  HvLabel,
  HvInfoMessage,
  HvFormElement,
  HvBaseDropdown,
  useControlled,
  HvTypography,
  useUniqueId,
  setId,
} from "@hitachivantara/uikit-react-core";
import { Checkbox } from "@hitachivantara/uikit-react-icons";

import styles from "./styles";
import Picker from "./Picker";
import PresetColors from "./PresetColors";

/**
 * A color picker component which allows the user to select a color from a list of pre-defined colors or freely select one color via the Hue and Saturation.
 * It receives a color string in HEX format and outputs an HEX formatted color.
 */
const HvColorPicker = (props) => {
  const theme = useTheme();
  const {
    id,
    name,

    required = false,
    disabled = false,

    label,
    "aria-label": ariaLabel,
    "aria-labelledby": ariaLabelledBy,
    description,
    "aria-describedby": ariaDescribedBy,
    className,
    classes,
    value,
    onChange,
    defaultValue = theme.hv.palette.accent.acce1,

    expanded,
    defaultExpanded = false,
  } = props;

  const [isOpen, setIsOpen] = useControlled(expanded, Boolean(defaultExpanded));
  const [color, setColor] = useControlled(value, defaultValue);

  const handleToggle = (_evt, s) => {
    setIsOpen(s);
  };

  const handleSelect = (val) => {
    onChange?.(val.hex);
    setColor(val.hex);
  };

  const elementId = useUniqueId(id, "hvdropdown");

  const hasLabel = label != null;
  const hasDescription = description != null;

  const colorArray = useMemo(
    () => [
      theme.hv.palette.semantic.sema4,
      theme.hv.palette.semantic.sema3,
      "#F6941E",
      "#8DC63D",
      theme.hv.palette.semantic.sema1,
      theme.hv.palette.semantic.sema19,
      "#05A99C",
      "#01ADEF",
      theme.hv.palette.accent.acce2h,
      "#0155A5",
      "#2E3192",
      "#652C90",
      theme.hv.palette.semantic.sema6,
      "#EC018B",
      theme.hv.palette.atmosphere.atmo5,
      theme.hv.palette.accent.acce1,
    ],
    [theme]
  );

  const setFocusToContent = (containerRef) => {
    const inputs = containerRef?.getElementsByTagName("input");
    if (inputs?.length > 0) {
      inputs[0].focus();
    }
  };

  return (
    <HvFormElement
      id={id}
      name={name}
      disabled={disabled}
      required={required}
      className={clsx(className, classes.root)}
    >
      {(hasLabel || hasDescription) && (
        <div className={classes.labelContainer}>
          {hasLabel && (
            <HvLabel id={setId(elementId, "label")} label={label} className={classes.label} />
          )}

          {hasDescription && (
            <HvInfoMessage id={setId(elementId, "description")} className={classes.description}>
              {description}
            </HvInfoMessage>
          )}
        </div>
      )}
      <HvBaseDropdown
        className={className}
        expanded={isOpen}
        onToggle={handleToggle}
        onContainerCreation={setFocusToContent}
        placeholder={
          <>
            <Checkbox className={classes.headerColorIcon} color={[color]} />
            <HvTypography className={classes.headerColorValue}>{color}</HvTypography>
          </>
        }
        aria-label={ariaLabel}
        aria-labelledby={
          [label && setId(elementId, "label"), ariaLabelledBy].join(" ").trim() || undefined
        }
        aria-describedby={
          [description && setId(elementId, "description"), ariaDescribedBy].join(" ").trim() ||
          undefined
        }
      >
        <HvPanel className={classes.panel}>
          <div className={classes.colorPicker}>
            <Picker color={color} onChange={handleSelect} />
            <PresetColors colors={colorArray} onClick={handleSelect} />
          </div>
        </HvPanel>
      </HvBaseDropdown>
    </HvFormElement>
  );
};

HvColorPicker.propTypes = {
  /**
   * Id to be applied to the form element root node.
   */
  id: PropTypes.string,

  /**
   * The form element name.
   */
  name: PropTypes.string,

  /**
   * The label of the form element.
   *
   * The form element must be labeled for accessibility reasons.
   * If not provided, an aria-label or aria-labelledby must be provided instead.
   */
  label: PropTypes.node,
  /**
   * @ignore
   */
  "aria-label": PropTypes.string,
  /**
   * @ignore
   */
  "aria-labelledby": PropTypes.string,
  /**
   * Provide additional descriptive text for the form element.
   */
  description: PropTypes.node,
  /**
   * @ignore
   */
  "aria-describedby": PropTypes.string,

  /**
   * Indicates that the form element is disabled.
   */
  disabled: PropTypes.bool,
  /**
   * Indicates that user input is required on the form element.
   */
  required: PropTypes.bool,

  /**
   * Class names to be applied.
   */
  className: PropTypes.string,
  /**
   * A Jss Object used to override or extend the styles applied.
   */
  classes: PropTypes.shape({
    root: PropTypes.string,
    labelContainer: PropTypes.string,
    label: PropTypes.string,
    description: PropTypes.string,
    headerColorIcon: PropTypes.string,
    headerColorValue: PropTypes.string,
    panel: PropTypes.string,
    colorPicker: PropTypes.string,
  }).isRequired,

  /**
   * If `true` the color picker starts opened if `false` it starts closed.
   */
  expanded: PropTypes.bool,
  /**
   * When uncontrolled, defines the initial expanded state.
   */
  defaultExpanded: PropTypes.bool,

  /**
   * The callback fired when the value changes.
   */
  onChange: PropTypes.func,

  /**
   * The value color, in HEX format.
   */
  value: PropTypes.string,

  /**
   * When uncontrolled, defines the initial value state.
   */
  defaultValue: PropTypes.string,
};

export default withStyles(styles, { name: "HvColorPicker" })(HvColorPicker);
