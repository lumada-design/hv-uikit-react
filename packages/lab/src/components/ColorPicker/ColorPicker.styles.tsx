import { CSSInterpolation } from "@emotion/serialize";
import { theme } from "@hitachivantara/uikit-react-core";
import { HvColorPickerClasses } from "./colorPickerClasses";

export const styles: Partial<
  Record<keyof HvColorPickerClasses, CSSInterpolation>
> = {
  labelContainer: {
    display: "flex",
    alignItems: "flex-start",
  },
  label: {
    paddingBottom: "6px",
    display: "block",
  },
  headerColorValue: {
    textTransform: "uppercase",
  },
  headerColorIcon: {
    width: 24,
    "& svg": {
      marginLeft: 0,
    },
  },
  panel: {
    width: "100%",
    minWidth: theme.colorPicker.panelMinWidth,
    display: "flex",
    justifyContent: "center",
    padding: theme.colorPicker.panelPadding,
    border: theme.colorPicker.panelBorder,
    borderRadius: theme.colorPicker.panelBorderRadius,
  },
  colorPicker: {
    width: theme.colorPicker.colorPickerWidth,
  },
  colorPickerIcon: {
    position: "absolute",
    pointerEvents: "none",
    top: -1,
    right: -1,
  },
  dropdownRootIconOnly: {
    width: 32,
    height: 32,
  },
  headerColorIconOnly: {
    position: "absolute",
    pointerEvents: "none",
    top: -1,
    right: -1,
    width: 32,
    height: 32,
  },
  pickerFields: { paddingBottom: 20 },
};
