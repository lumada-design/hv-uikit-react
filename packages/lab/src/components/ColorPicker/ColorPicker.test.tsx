import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { HvColorPicker } from "./ColorPicker";

describe("ColorPicker", () => {
  it("should render correctly full color picker", () => {
    const { container } = render(
      <div style={{ width: "134px" }}>
        <HvColorPicker
          label="Color"
          onChange={(color) => console.log(color)}
          expanded={true}
        />
      </div>
    );

    expect(container).toMatchSnapshot();
  });

  it("should render correctly without saved colors color picker", () => {
    const { container } = render(
      <div style={{ width: "134px" }}>
        <HvColorPicker
          label="Color"
          defaultValue="#de2beb"
          onChange={(color) => console.log(color)}
          expanded={true}
          showSavedColors={false}
        />
      </div>
    );

    expect(container).toMatchSnapshot();
  });

  it("should render correctly recommended colors only color picker", () => {
    const { container } = render(
      <div style={{ width: "134px" }}>
        <HvColorPicker
          label="Color"
          defaultValue="#de2beb"
          onChange={(color) => console.log(color)}
          expanded={true}
          showSavedColors={false}
          showCustomColors={false}
        />
      </div>
    );

    expect(container).toMatchSnapshot();
  });

  it("should render correctly full color icon picker", () => {
    const { container } = render(
      <div style={{ width: "134px" }}>
        <HvColorPicker
          label="Color"
          iconOnly={true}
          onChange={(color) => console.log(color)}
        />
      </div>
    );

    expect(container).toMatchSnapshot();
  });

  it("should render correctly without saved colors icon color picker", () => {
    const { container } = render(
      <div style={{ width: "134px" }}>
        <HvColorPicker
          label="Color"
          iconOnly={true}
          onChange={(color) => console.log(color)}
          showSavedColors={true}
        />
      </div>
    );

    expect(container).toMatchSnapshot();
  });

  it("should render correctly recommended colors only icon color picker", () => {
    const { container } = render(
      <div style={{ width: "134px" }}>
        <HvColorPicker
          label="Color"
          defaultValue="#de2beb"
          iconOnly={true}
          showSavedColors={true}
          showCustomColors={true}
          onChange={(color) => console.log(color)}
        />
      </div>
    );

    expect(container).toMatchSnapshot();
  });
});
