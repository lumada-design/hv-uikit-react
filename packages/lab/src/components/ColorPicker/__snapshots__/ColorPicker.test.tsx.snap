// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ColorPicker > should render correctly full color icon picker 1`] = `
<div>
  <div
    style="width: 134px;"
  >
    <div
      class="HvColorPicker-root HvFormElement-root"
    >
      <div
        class="HvColorPicker-labelContainer css-1yp4ln"
      >
        <label
          class="HvColorPicker-label HvLabel-root HvTypography-root HvTypography-label css-1e6khw8"
          id="hvdropdown50-label"
        >
          Color
        </label>
      </div>
      <div
        class="HvBaseDropdown-root HvColorPicker-dropdownRootIconOnly css-qqldeu"
      >
        <div
          aria-expanded="false"
          aria-labelledby="hvdropdown50-label"
          class="HvBaseDropdown-anchor css-fa7ybv"
          role="combobox"
          tabindex="-1"
        >
          <div
            aria-labelledby="hvdropdown50-label"
            class="HvBaseDropdown-header css-syatow"
            role="textbox"
            tabindex="0"
          >
            <div
              class="HvBaseDropdown-selection css-16osxvy"
            />
            <div
              class="HvColorPicker-colorPickerIcon css-1rtiqvt"
              name="ColorPicker"
            >
              <svg
                focusable="false"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="color0"
                  d="M15.415.567a2.047 2.047 0 0 0-2.828 0l-2.121 2.121L9.05 1.274l-.707.707L14 7.638l.707-.707-1.414-1.415 2.121-2.121a2 2 0 0 0 .001-2.828zM12.587 4.81zm2.121-2.122-2.121 2.121-1.414-1.414 2.121-2.121a1 1 0 0 1 1.414 1.414zM0 13v3h3l9-9-3-3zm2.586 2H1v-1.585l8-8L10.586 7z"
                  fill="var(--uikit-colors-secondary)"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ColorPicker > should render correctly full color picker 1`] = `
<div>
  <div
    style="width: 134px;"
  >
    <div
      class="HvColorPicker-root HvFormElement-root"
    >
      <div
        class="HvColorPicker-labelContainer css-1yp4ln"
      >
        <label
          class="HvColorPicker-label HvLabel-root HvTypography-root HvTypography-label css-1e6khw8"
          id="hvdropdown1-label"
        >
          Color
        </label>
      </div>
      <div
        class="HvBaseDropdown-root css-1kxonj9"
      >
        <div
          aria-expanded="true"
          aria-labelledby="hvdropdown1-label"
          aria-owns="hvbasedropdown3-children-container"
          class="HvBaseDropdown-anchor css-fa7ybv"
          role="combobox"
          tabindex="-1"
        >
          <div
            aria-controls="hvbasedropdown3-children-container"
            aria-labelledby="hvdropdown1-label"
            class="HvBaseDropdown-header HvBaseDropdown-headerOpen HvBaseDropdown-headerOpenDown css-qn77fs"
            role="textbox"
            tabindex="0"
          >
            <div
              class="HvBaseDropdown-selection css-16osxvy"
            >
              <p
                class="HvBaseDropdown-placeholder HvTypography-root HvTypography-body css-1isf4j6"
              >
                Select color...
              </p>
            </div>
            <div
              class="HvColorPicker-colorPickerIcon css-1rtiqvt"
              name="ColorPicker"
            >
              <svg
                focusable="false"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="color0"
                  d="M15.415.567a2.047 2.047 0 0 0-2.828 0l-2.121 2.121L9.05 1.274l-.707.707L14 7.638l.707-.707-1.414-1.415 2.121-2.121a2 2 0 0 0 .001-2.828zM12.587 4.81zm2.121-2.122-2.121 2.121-1.414-1.414 2.121-2.121a1 1 0 0 1 1.414 1.414zM0 13v3h3l9-9-3-3zm2.586 2H1v-1.585l8-8L10.586 7z"
                  fill="var(--uikit-colors-secondary)"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ColorPicker > should render correctly recommended colors only color picker 1`] = `
<div>
  <div
    style="width: 134px;"
  >
    <div
      class="HvColorPicker-root HvFormElement-root"
    >
      <div
        class="HvColorPicker-labelContainer css-1yp4ln"
      >
        <label
          class="HvColorPicker-label HvLabel-root HvTypography-root HvTypography-label css-1e6khw8"
          id="hvdropdown44-label"
        >
          Color
        </label>
      </div>
      <div
        class="HvBaseDropdown-root css-1kxonj9"
      >
        <div
          aria-expanded="true"
          aria-labelledby="hvdropdown44-label"
          aria-owns="hvbasedropdown46-children-container"
          class="HvBaseDropdown-anchor css-fa7ybv"
          role="combobox"
          tabindex="-1"
        >
          <div
            aria-controls="hvbasedropdown46-children-container"
            aria-labelledby="hvdropdown44-label"
            class="HvBaseDropdown-header HvBaseDropdown-headerOpen HvBaseDropdown-headerOpenDown css-qn77fs"
            role="textbox"
            tabindex="0"
          >
            <div
              class="HvBaseDropdown-selection css-16osxvy"
            >
              <div
                class="HvColorPicker-headerColorIcon css-11agson"
                name="Checkbox"
              >
                <svg
                  focusable="false"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    class="color0"
                    d="M0 0h16v16H0z"
                    fill="#de2beb"
                  />
                  <path
                    class="color1"
                    d="M16 16H0V0h16zM1 15h14V1H1z"
                  />
                </svg>
              </div>
              <p
                class="HvColorPicker-headerColorValue HvTypography-root HvTypography-body css-1p2262w"
              >
                #de2beb
              </p>
            </div>
            <div
              class="HvColorPicker-colorPickerIcon css-1rtiqvt"
              name="ColorPicker"
            >
              <svg
                focusable="false"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="color0"
                  d="M15.415.567a2.047 2.047 0 0 0-2.828 0l-2.121 2.121L9.05 1.274l-.707.707L14 7.638l.707-.707-1.414-1.415 2.121-2.121a2 2 0 0 0 .001-2.828zM12.587 4.81zm2.121-2.122-2.121 2.121-1.414-1.414 2.121-2.121a1 1 0 0 1 1.414 1.414zM0 13v3h3l9-9-3-3zm2.586 2H1v-1.585l8-8L10.586 7z"
                  fill="var(--uikit-colors-secondary)"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ColorPicker > should render correctly recommended colors only icon color picker 1`] = `
<div>
  <div
    style="width: 134px;"
  >
    <div
      class="HvColorPicker-root HvFormElement-root"
    >
      <div
        class="HvColorPicker-labelContainer css-1yp4ln"
      >
        <label
          class="HvColorPicker-label HvLabel-root HvTypography-root HvTypography-label css-1e6khw8"
          id="hvdropdown58-label"
        >
          Color
        </label>
      </div>
      <div
        class="HvBaseDropdown-root HvColorPicker-dropdownRootIconOnly css-qqldeu"
      >
        <div
          aria-expanded="false"
          aria-labelledby="hvdropdown58-label"
          class="HvBaseDropdown-anchor css-fa7ybv"
          role="combobox"
          tabindex="-1"
        >
          <div
            aria-labelledby="hvdropdown58-label"
            class="HvBaseDropdown-header css-syatow"
            role="textbox"
            tabindex="0"
          >
            <div
              class="HvBaseDropdown-selection css-16osxvy"
            />
            <div
              class="HvColorPicker-headerColorIconOnly css-zlcxrp"
              name="Checkbox"
            >
              <svg
                focusable="false"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="color0"
                  d="M0 0h16v16H0z"
                  fill="#de2beb"
                />
                <path
                  class="color1"
                  d="M16 16H0V0h16zM1 15h14V1H1z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ColorPicker > should render correctly without saved colors color picker 1`] = `
<div>
  <div
    style="width: 134px;"
  >
    <div
      class="HvColorPicker-root HvFormElement-root"
    >
      <div
        class="HvColorPicker-labelContainer css-1yp4ln"
      >
        <label
          class="HvColorPicker-label HvLabel-root HvTypography-root HvTypography-label css-1e6khw8"
          id="hvdropdown22-label"
        >
          Color
        </label>
      </div>
      <div
        class="HvBaseDropdown-root css-1kxonj9"
      >
        <div
          aria-expanded="true"
          aria-labelledby="hvdropdown22-label"
          aria-owns="hvbasedropdown24-children-container"
          class="HvBaseDropdown-anchor css-fa7ybv"
          role="combobox"
          tabindex="-1"
        >
          <div
            aria-controls="hvbasedropdown24-children-container"
            aria-labelledby="hvdropdown22-label"
            class="HvBaseDropdown-header HvBaseDropdown-headerOpen HvBaseDropdown-headerOpenDown css-qn77fs"
            role="textbox"
            tabindex="0"
          >
            <div
              class="HvBaseDropdown-selection css-16osxvy"
            >
              <div
                class="HvColorPicker-headerColorIcon css-11agson"
                name="Checkbox"
              >
                <svg
                  focusable="false"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    class="color0"
                    d="M0 0h16v16H0z"
                    fill="#de2beb"
                  />
                  <path
                    class="color1"
                    d="M16 16H0V0h16zM1 15h14V1H1z"
                  />
                </svg>
              </div>
              <p
                class="HvColorPicker-headerColorValue HvTypography-root HvTypography-body css-1p2262w"
              >
                #de2beb
              </p>
            </div>
            <div
              class="HvColorPicker-colorPickerIcon css-1rtiqvt"
              name="ColorPicker"
            >
              <svg
                focusable="false"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="color0"
                  d="M15.415.567a2.047 2.047 0 0 0-2.828 0l-2.121 2.121L9.05 1.274l-.707.707L14 7.638l.707-.707-1.414-1.415 2.121-2.121a2 2 0 0 0 .001-2.828zM12.587 4.81zm2.121-2.122-2.121 2.121-1.414-1.414 2.121-2.121a1 1 0 0 1 1.414 1.414zM0 13v3h3l9-9-3-3zm2.586 2H1v-1.585l8-8L10.586 7z"
                  fill="var(--uikit-colors-secondary)"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ColorPicker > should render correctly without saved colors icon color picker 1`] = `
<div>
  <div
    style="width: 134px;"
  >
    <div
      class="HvColorPicker-root HvFormElement-root"
    >
      <div
        class="HvColorPicker-labelContainer css-1yp4ln"
      >
        <label
          class="HvColorPicker-label HvLabel-root HvTypography-root HvTypography-label css-1e6khw8"
          id="hvdropdown54-label"
        >
          Color
        </label>
      </div>
      <div
        class="HvBaseDropdown-root HvColorPicker-dropdownRootIconOnly css-qqldeu"
      >
        <div
          aria-expanded="false"
          aria-labelledby="hvdropdown54-label"
          class="HvBaseDropdown-anchor css-fa7ybv"
          role="combobox"
          tabindex="-1"
        >
          <div
            aria-labelledby="hvdropdown54-label"
            class="HvBaseDropdown-header css-syatow"
            role="textbox"
            tabindex="0"
          >
            <div
              class="HvBaseDropdown-selection css-16osxvy"
            />
            <div
              class="HvColorPicker-colorPickerIcon css-1rtiqvt"
              name="ColorPicker"
            >
              <svg
                focusable="false"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="color0"
                  d="M15.415.567a2.047 2.047 0 0 0-2.828 0l-2.121 2.121L9.05 1.274l-.707.707L14 7.638l.707-.707-1.414-1.415 2.121-2.121a2 2 0 0 0 .001-2.828zM12.587 4.81zm2.121-2.122-2.121 2.121-1.414-1.414 2.121-2.121a1 1 0 0 1 1.414 1.414zM0 13v3h3l9-9-3-3zm2.586 2H1v-1.585l8-8L10.586 7z"
                  fill="var(--uikit-colors-secondary)"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
